### ISSUE TITLE
Provide a way to exclude some params to be trained

### ISSUE THREAD

**Issue by fourson** [Lightning-AI/pytorch-lightning#19393]
### Description & Motivation

I have a lightning module, which has two submodules named as `self.model` and `self.loss` (both of them are `torch.nn.Module`). And I want to prevent `self.loss` printed as parameters before training, and freeze it by excluding it from the parameters.

### Pitch

The lightning module could be defined as:
```
class A(LightningModule):
def __init__(self, model, loss):
    super().__init__()
    self.model=model # model
    self.loss=loss # perceptual loss 
```
However, both `self.model` and `self.loss` are registered as the parameters of A (\ie, will be printed before trianing).
And I wish to exclude `self.loss` from the parameters.

### Alternatives

I have tried many ways such as manually calling `self.loss().eval()` in `self.training_step()'. However the `self.loss` is still printed as parameters before training.

### Additional context

Is there any hooks like `self.exclude_param_list=[self.loss]`? Thx.

cc @borda

**Comment by awaelchli** [comment]
@fourson `.eval()` won't help, this only changes the mode of the module. To freeze it you need to do `self.loss.requires_grad_(False)`. 

It is not possible to remove a parameter from a PyTorch nn.Module directly, and so Lightning can't provide that functionality. But you can use a trick if you really care about it: If you include your loss into a list, PyTorch won't associate it as a submodule: `self.loss = [loss]`. The drawback is that `self.loss` won't be moved to the GPU automatically.

**Comment by awaelchli** [comment]
@fourson What do you think about my comment above?
Btw I also opened #19468 which adds a column "Mode" to the summary that shows which layers are in training mode. Your frozen layers would show up as "eval" there. Maybe that's helpful in your case.

For the reasons outlined in my previous comment, I don't think we can provide a mechanism to "exclude" parameters and modules. I believe any hacks we could do would go against PyTorch's design.

**Comment by fourson** [comment]
> @fourson What do you think about my comment above? Btw I also opened #19468 which adds a column "Mode" to the summary that shows which layers are in training mode. Your frozen layers would show up as "eval" there. Maybe that's helpful in your case.
> 
> For the reasons outlined in my previous comment, I don't think we can provide a mechanism to "exclude" parameters and modules. I believe any hacks we could do would go against PyTorch's design.

Thanks for the comment. The "eval" flag shown in the summary could be very helpful for further usage.
